scene/resources/gradient.h

```diff
@@ -137,13 +196,17 @@ class Gradient : public Resource {
 		const Point &pointFirst = points[first];
 		const Point &pointSecond = points[second];
 
+		Color color1 = transform_color_space(pointFirst.color);
+		Color color2 = transform_color_space(pointSecond.color);
+
 		switch (interpolation_mode) {
 			case GRADIENT_INTERPOLATE_LINEAR: {
-				return pointFirst.color.lerp(pointSecond.color, (p_offset - pointFirst.offset) / (pointSecond.offset - pointFirst.offset));
-			} break;
+				Color interpolated = color1.lerp(color2, (p_offset - pointFirst.offset) / (pointSecond.offset - pointFirst.offset));
+				return inv_transform_color_space(interpolated);
+			}
 			case GRADIENT_INTERPOLATE_CONSTANT: {
-				return pointFirst.color;
-			} break;
+				return inv_transform_color_space(color1);
```

Comment:
I would personally define the necessary color points inside the cases and collapse the LINEAR and the default case into one.

Code looks good otherwise, I wonder if the OKLAB color transforms and inverse transforms can be made nicer though.